webapp/components/wpt-flags.html

```diff
@@ -0,0 +1,71 @@
+<!--
+Copyright 2018 The WPT Dashboard Project. All rights reserved.
+Use of this source code is governed by a BSD-style license that can be
+found in the LICENSE file.
+-->
+
+<link rel="import" href="../bower_components/polymer/polymer-element.html">
+<link rel="import" href="../bower_components/paper-item/paper-item.html">
+<link rel="import" href="../bower_components/paper-checkbox/paper-checkbox.html">
+
+<!--
+`<wpt-flags>` is a component for checking wpt.fyi feature flags.
+-->
+<dom-module id="wpt-flags">
+  <script>
+    const WPT_FYI_FEATURES = ['queryBuilder'];
+
+    const _wptFlags = (superClass, opt_writable) => class extends superClass {
+      static get is() {
+        return 'wpt-flags';
+      }
+
+      static get properties() {
+        const props = {};
+        for (const feature of WPT_FYI_FEATURES) {
+          const stored = localStorage.getItem(`features.${feature}`);
+          const value = stored && JSON.parse(stored);
+          props[feature] = {
+            type: Boolean,
+            readOnly: !opt_writable,
+            notify: true,
+            value: value,
+          };
+        }
+        return props;
+      }
+    }
+    const WPTFlags = (superClass) => _wptFlags(superClass, false);
+  </script>
+</dom-module>
+
+<dom-module id="wpt-flags-editor">
+  <template>
+    <paper-item>
+      <paper-checkbox checked="{{queryBuilder}}">
+        Query Builder component
+      </paper-checkbox>
+    </paper-item>
+  </template>
+
+  <script>
+    class WPTFlagsEditor extends _wptFlags(window.Polymer.Element, /*writable*/ true) {
+      static get is() { return 'wpt-flags-editor'; }
+
+      ready() {
+        super.ready();
+        for (const feature of WPT_FYI_FEATURES) {
+          this._createMethodObserver(`valueChanged(${feature}, '${feature}')`);
```

Comment:
The leading underscore made me a bit worried at first, but looks like it's documented: https://www.polymer-project.org/2.0/docs/devguide/observers

Reply:
I had the same reaction.